From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Sun, 2 Jul 2023 09:19:17 -0500
Subject: [PATCH] Add computSide api from Paper PR-9388

Deprecated as this will be removed as soon as merged upstream

diff --git a/src/main/java/com/empireminecraft/api/EAPI_Misc.java b/src/main/java/com/empireminecraft/api/EAPI_Misc.java
index 88ffd53ea085a68e65044cde9639a584b168f693..833d2fd2ee2880bb67259ae823271968165a515e 100644
--- a/src/main/java/com/empireminecraft/api/EAPI_Misc.java
+++ b/src/main/java/com/empireminecraft/api/EAPI_Misc.java
@@ -25,6 +25,8 @@ package com.empireminecraft.api;
 
 import org.bukkit.Location;
 import org.bukkit.StructureType;
+import org.bukkit.block.Sign;
+import org.bukkit.block.sign.Side;
 import org.bukkit.inventory.ItemStack;
 import org.jetbrains.annotations.NotNull;
 
@@ -38,4 +40,5 @@ public interface EAPI_Misc {
      */
     @NotNull
     ItemStack createTreasureMap(@NotNull StructureType type, @NotNull Location loc);
+
 }
diff --git a/src/main/java/org/bukkit/block/Sign.java b/src/main/java/org/bukkit/block/Sign.java
index dd4d8f24e684a9c3cb1e41e8d0bf615111a17ba3..8f453bd98d0cb8e9772a5b9bee73f4e171f1bf39 100644
--- a/src/main/java/org/bukkit/block/Sign.java
+++ b/src/main/java/org/bukkit/block/Sign.java
@@ -174,4 +174,16 @@ public interface Sign extends TileState, Colorable {
      */
     @NotNull
     public SignSide getSide(@NotNull Side side);
+    // Paper start
+    @Deprecated
+    default @NotNull Side computeVisibleSide(org.bukkit.entity.@NotNull Entity entity) {
+        return this.computeVisibleSide(entity.getLocation());
+    }
+    @Deprecated
+    default @NotNull Side computeVisibleSide(io.papermc.paper.math.@NotNull Position position) {
+        return this.computeVisibleSide(position.x(), position.z());
+    }
+    @Deprecated
+    @NotNull Side computeVisibleSide(double x, double z);
+    // Paper end
 }
